### PR TITLE
BAPL-1148: align to JBEAP 7.2

### PR DESCRIPTION
**Pull request by mbiarnes**
changed jboss-ip-bom
upgraded to org.dom4j
upgraded org.jboss.spec.javax.ws.rs:jboss-jaxrs-api_2.1_spec
(cherry picked from commit 4d927acdb9c73c8642e39b6e9369040909a3ab82)

### CODE REVIEW

**Comment by mbiarnes** [comment]
merge together..:
https://github.com/errai/errai/pull/378
https://github.com/kiegroup/kie-soup/pull/82
https://github.com/kiegroup/appformer/pull/648
https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/933
https://github.com/kiegroup/optaplanner/pull/488
https://github.com/kiegroup/jbpm/pull/1450
https://github.com/kiegroup/droolsjbpm-integration/pull/1749
https://github.com/kiegroup/kie-wb-common/pull/2501
https://github.com/kiegroup/jbpm-wb/pull/1305
https://github.com/kiegroup/jbpm-work-items/pull/97
https://github.com/kiegroup/kie-docs/pull/1414
https://github.com/kiegroup/kie-wb-distributions/pull/883
https://github.com/kiegroup/optaweb-employee-rostering/pull/229

**Comment by kiereleaseuser2** [comment]
Build finished. 2761 tests run, 7 skipped, 7 failed.

**Comment by tiagobento** [comment]
@mbiarnes We have 7 tests failing, can you check that?

**Comment by mbiarnes** [comment]
@tiagobento https://rhba-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/KIE/job/master/job/pullrequest/job/errai-pullrequests/lastCompletedBuild/testReport/

**Comment by mbiarnes** [comment]
Please do not merge yet. There are test failures. @tiagobento are fining them. He should decide when this PR can  be merged.

**Comment by tiagobento** [comment]
@mbiarnes Just opened a PR on your fork to add to this PR. Please merge it https://github.com/mbiarnes/errai/pull/1 and trigger Jenkins again.

**Comment by mbiarnes** [comment]
@tiagobento ThanX.

**Comment by mbiarnes** [comment]
Jenkins please retest this

File: errai-jaxrs/errai-jaxrs-client/src/test/java/org/jboss/errai/enterprise/jaxrs/client/shared/QueryParamTestService.java [hunk 1]
```diff
@@ -59,7 +59,11 @@
   @GET
   @Path("/5")
   public Set<String> getWithQueryParamSetOfStrings(@QueryParam("id") Set<String> id);
-  
+
+  @GET
+  @Path("/5t")
+  public Set<String> getWithQueryParamSetOfStringsTreeSet(@QueryParam("id") Set<String> id);
+
```
Comment (mareknovotny) [inline errai-jaxrs/errai-jaxrs-client/src/test/java/org/jboss/errai/enterprise/jaxrs/client/shared/QueryParamTestService.java]: why is this here in alignment PR?

File: errai-jaxrs/errai-jaxrs-client/src/test/java/org/jboss/errai/enterprise/jaxrs/client/test/QueryParamIntegrationTest.java [hunk 2]
```diff
@@ -96,6 +97,13 @@ public void testGetWithQueryParamSetOfStrings() {
         new SimpleAssertionCallback<>("@GET with Set<String> as @QueryParams failed", strings)).getWithQueryParamSetOfStrings(strings);
   }
 
+  @Test
+  public void testGetWithQueryParamSetOfStringsTreeSet() {
+    final Set<String> strings = new TreeSet<>(Arrays.asList("1", "2", "3"));
+    call(QueryParamTestService.class,
+            new SimpleAssertionCallback<>("@GET with Set<String> as @QueryParams failed", strings)).getWithQueryParamSetOfStringsTreeSet(strings);
+  }
+
```
Comment (mareknovotny) [inline errai-jaxrs/errai-jaxrs-client/src/test/java/org/jboss/errai/enterprise/jaxrs/client/test/QueryParamIntegrationTest.java]: why is this here in alignment PR?

File: errai-jaxrs/errai-jaxrs-client/src/test/java/org/jboss/errai/enterprise/jaxrs/server/QueryParamTestServiceImpl.java [hunk 3]
```diff
@@ -55,6 +56,11 @@ public String getWithMultipleQueryParams(long id1, long id2) {
     return id;
   }
 
+  @Override
+  public Set<String> getWithQueryParamSetOfStringsTreeSet(Set<String> id) {
+    return new TreeSet<>(id);
+  }
+
```
Comment (mareknovotny) [inline errai-jaxrs/errai-jaxrs-client/src/test/java/org/jboss/errai/enterprise/jaxrs/server/QueryParamTestServiceImpl.java]: why is this here in alignment PR?

File: errai-jaxrs/errai-jaxrs-client/src/test/java/org/jboss/errai/enterprise/jaxrs/client/shared/QueryParamTestService.java (hunk 1 above)
Comment (mbiarnes) [reply errai-jaxrs/errai-jaxrs-client/src/test/java/org/jboss/errai/enterprise/jaxrs/client/shared/QueryParamTestService.java]: brought in by the PR @tiagobento raised to fix failing tests in errai

File: errai-jaxrs/errai-jaxrs-client/src/test/java/org/jboss/errai/enterprise/jaxrs/client/test/QueryParamIntegrationTest.java (hunk 2 above)
Comment (mbiarnes) [reply errai-jaxrs/errai-jaxrs-client/src/test/java/org/jboss/errai/enterprise/jaxrs/client/test/QueryParamIntegrationTest.java]: brought in by the PR @tiagobento raised to fix failing tests in errai

File: errai-jaxrs/errai-jaxrs-client/src/test/java/org/jboss/errai/enterprise/jaxrs/server/QueryParamTestServiceImpl.java (hunk 3 above)
Comment (mbiarnes) [reply errai-jaxrs/errai-jaxrs-client/src/test/java/org/jboss/errai/enterprise/jaxrs/server/QueryParamTestServiceImpl.java]: brought in by the PR @tiagobento raised to fix failing tests in errai

**Comment by kiereleaseuser2** [comment]
Build finished. 2762 tests run, 7 skipped, 0 failed.

**Comment by mbiarnes** [comment]
@tiagobento can you please look at: [errai](https://rhba-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/KIE/job/master/job/pullrequest/job/errai-pullrequests/58/console), the build failed in Errai::UI::Demos::I18NDemo

File: errai-jaxrs/errai-jaxrs-client/src/test/java/org/jboss/errai/enterprise/jaxrs/client/shared/QueryParamTestService.java (hunk 1 above)
Comment (mareknovotny) [reply errai-jaxrs/errai-jaxrs-client/src/test/java/org/jboss/errai/enterprise/jaxrs/client/shared/QueryParamTestService.java]: ok

**Comment by tiagobento** [comment]
@mbiarnes Ok, I'll take a look as soon as I can.

**Comment by mbiarnes** [comment]
Jenkins please retest this

**Comment by tiagobento** [comment]
@mbiarnes Just opened a second PR on your fork.

- I created a separate module called `errai-jaxrs-resteasy` with the configuration specific to the RestEasy implementation of JAX-RS. **Modules depending on both `errai-jaxrs-provider` _and_ the new RestEasy version MUST have `errai-jaxrs-resteasy` as a dependency**, or else `errai-marshalling` won't work properly. So please take a look at the related PRs and change them when needed.
This was necessary because the configuration I added on the previous commit broke GWT compilation when there was no RestEasy involved. Errai Marshalling needs the classes declared as mapping alises to be present on the classpath during compilation.

- I updated the version of the JAX-RS API on `maven-dependency-plugin` on `errai-jaxrs-client` as well so that it conforms with the new API. @porcelli can you please double-check the license headers of the files on the package `org.jboss.errai.enterprise.compat.javax.ws.rs.core`, please?

**Comment by mbiarnes** [comment]
thanX.
Merged. Lets see what happens :)

**Comment by kiereleaseuser2** [comment]
Build finished. 2762 tests run, 7 skipped, 0 failed.

**Comment by mbiarnes** [comment]
thanX @tiagobento 
the build was green: https://rhba-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/KIE/job/master/job/pullrequest/job/errai-pullrequests/59/console

**Comment by tiagobento** [comment]
🎉

**Comment by mareknovotny** [comment]
@gatekeepers please merge this
